### PR TITLE
Swift 4.2 support

### LINKED ISSUE #73: Support Swift 4.2

### PR DESCRIPTION
**Pull request by theblixguy**
This PR adds support for Swift 4.2. Closes #73

### CODE REVIEW

**Comment by theblixguy** [comment]
Seems like the build is failing because the CI is not using Xcode 10?

File: .swiftlint.yml [hunk 1]
```diff
@@ -1,3 +1,5 @@
 disabled_rules:
   - line_length
   - nesting
+identifier_name:
+  min_length: 1
```
Comment (KaneCheshire) [inline .swiftlint.yml]: What was the motivation behind adding this change? Were we getting an error from something?

**Comment by KaneCheshire** [comment]
We'll need to update the changelog as well I think? Probably means bumping the version to a new major version since this is a breaking change (I really wish Apple also used semantic versioning for Swift, since a source code change should really not be a Swift 4.1 to 4.2 version change!)

File: .swiftlint.yml (hunk 1 above)
Comment (theblixguy) [reply .swiftlint.yml]: Yeah, the tests on the CI were failing, so I had to make this change. The min length is 3 (default), but there were some variables in the tests that were of length 1 and SwiftLint didn't like that and it caused the build to fail.

**Comment by theblixguy** [comment]
> We'll need to update the changelog as well I think? Probably means bumping the version to a new major version since this is a breaking change (I really wish Apple also used semantic versioning for Swift, since a source code change should really not be a Swift 4.1 to 4.2 version change!)

Done

File: .swiftlint.yml (hunk 1 above)
Comment (KaneCheshire) [reply .swiftlint.yml]: What were the variables out of interest? I'd rather fix those if we can

Comment (theblixguy) [reply .swiftlint.yml]: The variables are located in the file `TABResourceLoader/Concrete types/Result.swift`, line 50 & 51:

```
case .success(let v): return v
case .failure(let e): throw e
```

The names`v` and `e` need to be updated so they are at least 3 or more characters in length.

Here's what the CI log says:

```
Identifier Name Violation: Variable name should be between 3 and 40 characters long: 'v' (identifier_name)
Identifier Name Violation: Variable name should be between 3 and 40 characters long: 'e' (identifier_name)
```

Comment (KaneCheshire) [reply .swiftlint.yml]: Yeah I think we should strive for best practices and not just disable rules to get round them ;) 

Can we change those variables to be something a bit more descriptive? I have no idea what v is even meant to stand for here!

Maybe `v` should actually be something like `success` or `model` and then `e` be `error`?

Comment (theblixguy) [reply .swiftlint.yml]: Fixed it!

File: Sources/TABResourceLoader/Concrete types/Result.swift [hunk 2]
```diff
@@ -47,8 +47,8 @@ extension Result {
   // Convert the `Result` back to typical Swift `throws` error handling
   public func unwrap() throws -> T {
     switch self {
-    case .success(let v): return v
-    case .failure(let e): throw e
+    case .success(let capturedValue): return capturedValue
```
Comment (KaneCheshire) [inline Sources/TABResourceLoader/Concrete types/Result.swift:50]: 👌  much better imo

**Review by KaneCheshire — APPROVED** [review]

**Comment by KaneCheshire** [comment]
Let me know when this is merged and we can look at pushing the update to Cocoapods. I'm not sure if I have ownership rights over the pod though 🤔

**Comment by theblixguy** [comment]
> Let me know when this is merged and we can look at pushing the update to Cocoapods. I'm not sure if I have ownership rights over the pod though 🤔

I'll merge it now 😄

**Pull request merged**